source/main/utils/handleCheckBlockReplayProgress.js

```diff
@@ -20,19 +55,26 @@ export const handleCheckBlockReplayProgress = (
     const rl = readline.createInterface({ input: fileStream });
     const progress = [];
     for await (const line of rl) {
-      if (line.includes('block replay')) {
+      if (containProgressKeywords(line)) {
         progress.push(line);
       }
     }
     if (!progress.length) return;
 
     const finalProgress = progress.slice(-1).pop();
-    const percentage = finalProgress.split('block replay progress (%) =').pop();
+    const match = finalProgress.match(/Progress:([\s\d.,]+)%/);
+    const progressType = getProgressType(finalProgress);
+    if (!match || !progressType) {
+      return;
+    }
+    const [, percentage] = match;
```

Comment:
A cosmetic suggestion:
```suggestion
    const percentage = finalProgress.match(/Progress:([\s\d.,]+)%/)?.[1];
    const progressType = getProgressType(finalProgress);
    if (!percentage || !progressType) {
      return;
    }
```